### PR TITLE
NIFI-5292 Renamed ElasticSearch client service impl to show it is for…

### PR DESCRIPTION
**Pull request by MikeThomsen**
… 5x.

NIFI-5292 Added 6.X ES Client.

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by MikeThomsen** [comment]
Looked over the transitive dependencies and HdrHistogram ([license info; appears public domain](https://github.com/HdrHistogram/HdrHistogram/blob/master/LICENSE.txt)) and SnakeYaml (ASL, unknown copyright) appear to be the only two needed to be added to the NOTICE for the v6 version.

**Comment by MikeThomsen** [comment]
@markap14 can you review?

@joewitt can you review the L&N? I took a stab at updating the v6 client's NOTICE, but am not sure if it's right.

**Comment by markap14** [comment]
Hey @MikeThomsen when we discussed the renaming on the mailing list, I was under the impression that the Elasticsearch Client Service was new for 1.7.0. But it appears that it already has been released in 1.6.0. We need to avoid changing the names of components once they have been released because doing so would break users' flows. I think we need to keep the name nifi-elasticsearch-client-service and then we can use the name nifi-elasticsearch-client-service-6, etc. for the new one. Then we'd just document the old one to indicate that it was designed to work against Elasticsearch 5.x. Sorry for the confusion! We do need to ensure that we keep existing flows stable, though, and we only want to break a flow if there's no feasible way around it.

**Comment by MikeThomsen** [comment]
@markap14 Ok, that's fair. I think the general issue has to be addressed, but the easy route is obviously blocked. So what I'm going to do to try to head this is off is:

1. Rip out the changes to the 5x client and keep the 6x version.
2. Add some metadata to the 5x client service to help provide some visual cues in the UI.
3. See if there is a way to get the protocol compatibility range from the ES client API. I've seen debug info that suggests it looks for that and enforces it, so I'm hoping there's a client API which will let us sniff that and throw an error saying you need to change client services to keep w/ ES protocol compatibility.

**Comment by MikeThomsen** [comment]
@markap14 I cleaned it up and got most of what I suggested done. Couldn't figure out a good way to detect the protocol ranges, but this should do.

**Comment by MikeThomsen** [comment]
@zenfenan can you review? It's mostly copy pasta from the ES v5 client.

**Comment by mattyb149** [comment]
@MikeThomsen Did you close this due to inactivity, or did you move the changes into a different PR, or something else?

**Comment by MikeThomsen** [comment]
I don't even remember closing it, that's the weird part.

File: nifi-assembly/pom.xml [hunk 1]
```diff
@@ -475,6 +475,12 @@ language governing permissions and limitations under the License. -->
             <version>1.9.0-SNAPSHOT</version>
             <type>nar</type>
         </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-elasticsearch-client-service-6x-nar</artifactId>
+            <version>1.8.0-SNAPSHOT</version>
```
Comment (mattyb149) [inline nifi-assembly/pom.xml]: These should all be updated to 1.11.0-SNAPSHOT, that might be what's causing the non-cached Maven build failures in Travis.

**Comment by MikeThomsen** [comment]
Going to follow @markap14 's original guidance and close this as it would constitute a breaking change. Plus, we don't technically need this since the most recent changes to this bundle merged in preliminary support for ES 7

**Pull request closed without merging**